internal/agent/agent.go

```diff
@@ -274,24 +276,53 @@ func checkCollectorConnectivity(ctx context2.Context, cfg *config.Config, retrie
 	return
 }
 
-func getSampleMatcher(c *config.Config) func(interface{}) bool {
+func newSampleMatcher(c *config.Config, ffRetriever feature_flags.Retriever) func(interface{}) bool {
```

Comment:
This function start to be complex and difficult to read, it is possible to refactor? 🤔 
 
Sounds there are diff conditions to "build" the returned function.

Reply:
I cleaned this up a bit, let me know..

Reply:
Cool refactor!